libavpipe/src/avpipe_xc.c

```diff
@@ -4921,13 +4927,31 @@ avpipe_fini(
         }
     }
 
+    if ((*xctx)->params->copy_mpegts) {
+        cp_ctx_t *cp_ctx = &(*xctx)->cp_ctx;
+        coderctx_t *mpegts_encoder_ctx = &cp_ctx->encoder_ctx;
+        elv_warn("Freeing mpegts encoder context, closing io");
+        avio_close(mpegts_encoder_ctx->format_context->pb);
```

Comment:
Should we be using this function in other places too (like above in this function)?

Reply:
I have to be totally honest and say that I'm not really sure. I used it because it seemed appropriate, but how exactly to close everything is not super well documented, and I'm not sure if it is necessary or not.